src/main/java/nl/opengeogroep/safetymaps/server/stripes/FotoFunctionActionBean.java

```diff
@@ -0,0 +1,200 @@
+    public void setContext(ActionBeanContext context) {
+        this.context = context;
+    }
+
+    @Override
+    public ActionBeanContext getContext() {
+        return context;
+    }
+
+    public FileBean getPicture() {
+        return picture;
+    }
+
+    public void setPicture(FileBean picture) {
+        this.picture = picture;
+    }
+
+    @DefaultHandler
+    public Resolution foto() throws IOException, ServletException {
+        JSONObject response = new JSONObject();
+        try {
+            PATH = Cfg.getSetting("fotofunctie");
+
+            response.put("result", false);
+            if (extraInfo == null) {
+                extraInfo = "";
+            }
+
+            if (incidentNummer == null) {
+                incidentNummer = "N.V.T.";
+            }
+
+            insertIntoDb();
+            final File file = new File(PATH + fileName);
+            picture.save(file);
+            setPermissions();
+            response.put("message", "Foto is opgeslagen met bestandsnaam: " + fileName);
+            response.put("result", true);
+        } catch (Exception e) {
+            response.put("message", "Error met fout " + e.getMessage());
+        }
+        return new StreamingResolution("application/json", response.toString());
+    }
+
+    public void setPermissions() throws IOException {
+        Set<PosixFilePermission> perms = new HashSet<PosixFilePermission>();
+        //add owners permission
+        perms.add(PosixFilePermission.OWNER_READ);
+        perms.add(PosixFilePermission.OWNER_WRITE);
+        perms.add(PosixFilePermission.OWNER_EXECUTE);
+        //add group permissions
+        perms.add(PosixFilePermission.GROUP_READ);
+        perms.add(PosixFilePermission.GROUP_WRITE);
+        perms.add(PosixFilePermission.GROUP_EXECUTE);
+        //add others permissions
+        perms.add(PosixFilePermission.OTHERS_READ);
+        perms.add(PosixFilePermission.OTHERS_WRITE);
+        perms.add(PosixFilePermission.OTHERS_EXECUTE);
+
+        Files.setPosixFilePermissions(Paths.get(PATH + fileName), perms);
+    }
+
+    public Resolution fotoForIncident() throws Exception {
+        JSONArray response = new JSONArray();
+
+        List<Map<String, Object>> rows = getFromDb();
+
+        for (Map<String, Object> row : rows) {
+            response.put(rowToJson(row, false, false));
+        }
+
+        return new StreamingResolution("application/json", response.toString());
+    }
+
+    public void insertIntoDb() throws Exception {
+        Calendar calendar = Calendar.getInstance();
+        java.sql.Date date = new java.sql.Date(calendar.getTime().getTime());
+        QueryRunner qr = DB.qr();
+        qr.insert("insert into wfs."+TABLE+" (filename, datatype, voertuig_nummer, incident_nummer,location, date, omschrijving) values('" + fileName + "','" + type + "','" + voertuigNummer + "','" + incidentNummer + "','" + PATH + fileName + "','" + date + "','" + extraInfo + "') ", new MapListHandler());
```

Comment:
SQL injection

It is better to not save the PATH in the database but only the filename, so the directory can be moved